spack/packages/dla-future/package.py

```diff
@@ -149,6 +149,9 @@ def cmake_args(self):
             args.append(self.define("BUILD_TESTING", True))
             args.append(self.define("DLAF_BUILD_TESTING", True))
             args.append(self.define("DLAF_CI_RUNNER_USES_MPIRUN", True))
+
+            if "~cuda" in self.spec and "~rocm" in self.spec:
+                args.append(self.define("DLAF_TEST_PREFLAGS", "check-threads"))
```

Comment:
I added a `DLAF_TEST_PREFLAGS` CMake variable which is included in between MPI flags (if any) and the test executable. I use this to add the `check-threads` wrapper script. For symmetry I also added the option for `DLAF_TEST_POSTFLAGS`, but this is unused.

I currently have the logic for choosing to use `check-threads` into the spack package itself. There are (at least) two other options:

1. Inject `DLAF_TEST_PREFLAGS` the same way the coverage flag is injected, i.e. by rewriting the package file in CI.
2. Add a variant to enable this (`variant("ci-check-threads", ...)`?)
3. ???

Would you prefer any of these over the current implementation?

Reply:
I'm starting to think that we should add these "developer" variants in the spack package in our repository, and only remove them when we upload it to the spack repo.
In this way the full set of variants is available, if we want to do a quick manual test on any system, and we avoid package rewriting workaround in the CI. These variants will not be exposed to the users as they will mainly use the package from the spack repo.

Reply:
I'd be in favour of this. We're not going to run out of variants, so no need to stuff all the options behind `+ci-test`.